crates/network-scheduler/src/worker_state.rs

```diff
@@ -118,6 +123,25 @@ impl WorkerState {
         self.stored_bytes = msg.stored_bytes.unwrap_or_default();
     }
 
+    pub fn heartbeat(&mut self, msg: Heartbeat) {
+        self.last_ping = Some(SystemTime::now());
+        self.version = Some(msg.version);
+        if let Some(missing_chunks) = msg.missing_chunks {
+            let mut decoder = DeflateDecoder::new(&missing_chunks.data[..]);
+            let mut unavailability_map = Vec::<u8>::new();
+            _ = decoder.read_to_end(&mut unavailability_map);
+            self.num_missing_chunks_on_heartbeat = Some(unavailability_map.iter().filter(|v| **v > 0).count() as u32);
```

Comment:
Let's move it to a common crate: https://github.com/subsquid/sqd-network/pull/129

Reply:
done